File: starter_kits/Rust/src/hlt/map_cell.rs
use hlt::DropoffId;
use hlt::PlayerId;
use hlt::position::Position;

pub struct MapCell {
    pub position: Position,
    pub halite: usize,
    pub structure: Structure,
}

#[derive(Eq, PartialEq)]
pub enum Structure {
    None,
    Dropoff(DropoffId),
    Shipyard(PlayerId),
}

impl Structure {
    pub fn is_some(&self) -> bool {
        match *self {
            Structure::None => false,
            _ => true,
        }
    }

    pub fn is_none(&self) -> bool {
        !self.is_some()
    }
}

File: starter_kits/PHP/hlt/MapCell.php
<?php

namespace hlt;

class MapCell
{
    /**
     * @var Position
     */
    public $position;
    /**
     * @var int
     */
    public $halite;
    /**
     * @var Ship
     */
    public $ship;
    /**
     * @var Entity
     */
    public $structure;

    public function __construct(Position $position, int $halite)
    {
        $this->position = $position;
        $this->halite = $halite;
    }

    public function isEmpty(): bool
    {
        return $this->ship === null && $this->structure === null;
    }

    public function isOccupied(): bool
    {
        return $this->ship !== null;
    }

    public function hasStructure(): bool
    {
        return $this->structure !== null;
    }

    public function markUnsafe(Ship $ship): void
    {
        $this->ship = $ship;
    }
}
